### DISCUSSION TITLE
write error can't recover even the error condition is gone

### DISCUSSION THREAD

**Discussion by qingyunha** [juicedata/juicefs#2392] [General]
I notice `fileWriter.Write` return `fileWriter.err` and `fileWriter.err` never clear after it be assigned a errror value.

**Comment by davies** [comment]
When there is an error during writing, which means some data is already lost, it can't recover the lost data so the error should not be cleared.

**Comment by qingyunha** [comment]
Considing this scenario, a long runing process that attempts to write to a log file but encounters a no space left error should succeed after the disk has enough space.
